app/models/request.rb

```diff
@@ -7,4 +7,20 @@ class Request < ApplicationRecord
   validates :content, :user, :duedate, presence: true
 
   enum status: { pending: 0, accepted: 1, rejected: 2, commented: 3 }
+
+  def check_status
+    check = false
+
+    receivers.each do |receiver|
+      if receiver.status == 'accepted'
```

Comment:
what about `receivers.all?(&:accepted?)`
if it returns true, then you can change the status.

something like this:
 `accepted! if receivers.all?(&:accepted?)`